### PR TITLE
feat: support init global table and recover table when tablet restart 

### PR DESCRIPTION
**Pull request by keyu813**
1. support init global system table when nameserver init
2. support recover system table when tablet restart
3. add global var `sync_job` `job_timeout`

### CODE REVIEW

File: src/nameserver/name_server_impl.cc [hunk 1]
```diff
@@ -1278,6 +1285,33 @@ void NameServerImpl::RecoverEndpointInternal(const std::string& endpoint, bool n
...
         RecoverEndpointDBInternal(endpoint, need_restore, concurrency, kv.second);
     }
+    // recover global variable after tablet restart
+    std::shared_ptr<TableInfo> table_info;
+    if (!GetTableInfo(GLOBAL_VARIABLES, INFORMATION_SCHEMA_DB, &table_info)) {
+        PDLOG(WARNING, "table is not exist!");
+        return;
+    }
+    bool exist_globalvar = false;
+    for (int meta_idx = 0; meta_idx < table_info->table_partition(0).partition_meta_size(); meta_idx++) {
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc:1295]: should check all partition

File: src/nameserver/name_server_impl.cc [hunk 2]
```diff
@@ -1278,6 +1285,33 @@ void NameServerImpl::RecoverEndpointInternal(const std::string& endpoint, bool n
...
+
+void NameServerImpl::NotifyGlobalVarChanged() {
+    if (!IsClusterMode()) {
+        return;
+    }
+    if (!zk_client_->Increment(zk_path_.globalvar_changed_notify_node_)) {
+        PDLOG(WARNING, "increment failed, node is %s", zk_path_.globalvar_changed_notify_node_);
+        return;
+    }
+    PDLOG(INFO, "notify table changed ok");
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc]: update log msg

File: src/nameserver/name_server_impl.cc [hunk 3]
```diff
@@ -10267,5 +10309,57 @@ void NameServerImpl::UpdateOfflineTableInfo(::google::protobuf::RpcController* c
...
+        }
+        // system table only have one partition, so table_partition(0) can be used
+        for (int meta_idx = 0; meta_idx < table_info->table_partition(0).partition_meta_size(); meta_idx++) {
+            if (table_info->table_partition(0).partition_meta(meta_idx).is_leader()) {
+                uint64_t cur_ts = ::baidu::common::timer::get_micros() / 1000;
+                std::string endpoint = table_info->table_partition(0).partition_meta(meta_idx).endpoint();
+                auto table_ptr = GetTablet(endpoint);
+                if (!table_ptr->client_->Put(tid, pid, cur_ts, row, dimensions)) {
+                    LOG(WARNING) << "fail to make a put request to table. tid " + std::to_string(tid);
+                    return {};
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc]: `{}` will use default constructor and set as ok. if some error occurs, error code and msg should be set.

File: src/nameserver/name_server_impl.cc [hunk 4]
```diff
@@ -1279,6 +1286,35 @@ void NameServerImpl::RecoverEndpointInternal(const std::string& endpoint, bool n
     for (const auto& kv : db_table_info_) {
         RecoverEndpointDBInternal(endpoint, need_restore, concurrency, kv.second);
     }
+    // recover global variable after tablet restart
+    std::shared_ptr<TableInfo> table_info;
+    if (!GetTableInfoUnlock(GLOBAL_VARIABLES, INFORMATION_SCHEMA_DB, &table_info)) {
+        PDLOG(WARNING, "table is not exist!");
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc]: add table name

File: src/nameserver/name_server_impl.cc [hunk 5]
```diff
@@ -10268,5 +10309,59 @@ void NameServerImpl::UpdateOfflineTableInfo(::google::protobuf::RpcController* c
...
+    for (int i = 0; i < default_value.size(); i++) {
+        std::string row = rows[i];
+        std::vector<std::pair<std::string, uint32_t>> dimensions = rows_dimensions[i];
+        uint32_t pid = 0;
+        if (pid_num > 0) {
+            pid = (uint32_t)(::openmldb::base::hash64(dimensions[0].first) % pid_num);
+        }
+        // system table only have one partition, so table_partition(0) can be used
+        for (int meta_idx = 0; meta_idx < table_info->table_partition(0).partition_meta_size(); meta_idx++) {
+            if (table_info->table_partition(0).partition_meta(meta_idx).is_leader()) {
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc]: check leader and is_alive is true

File: src/nameserver/name_server_impl.cc [hunk 6]
```diff
@@ -10268,5 +10309,59 @@ void NameServerImpl::UpdateOfflineTableInfo(::google::protobuf::RpcController* c
...
+                uint64_t cur_ts = ::baidu::common::timer::get_micros() / 1000;
+                std::string endpoint = table_info->table_partition(0).partition_meta(meta_idx).endpoint();
+                auto table_ptr = GetTablet(endpoint);
+                if (!table_ptr->client_->Put(tid, pid, cur_ts, row, dimensions)) {
+                    return {ReturnCode::kPutFailed, "fail to make a put request to table"};
+                }
```
Comment (dl239) [inline src/nameserver/name_server_impl.cc:10360]: add `break` here

**Review by dl239 — APPROVED** [review]

**Review by aceforeverd — APPROVED** [review]

**Comment by keyu813** [comment]
[global variable support](https://trello.com/c/Nqn4zAcQ/29-global-variable-support)

**Pull request merged**